docs/src/main/asciidoc/class-loading-reference.adoc

```diff
@@ -19,6 +19,30 @@ a normal production Quarkus application.
 For all other use cases (e.g. tests, dev mode, and building the application) Quarkus
 uses the class loading architecture outlined here.
 
+== Reading Class Bytecode
+
+Whenever class bytecode needs to be read, a `@BuildStep` where class reading is done have
+to have `loadsApplicationClasses` parameter set to `true`. Otherwise the class will not be
+found.
```

Comment:
```suggestion
Whenever class bytecode needs to be read, a `@BuildStep` where class reading is done must have the `loadsApplicationClasses` parameter set to `true`. Otherwise the class will not be found.
```